src/symbol/placement.js

```diff
@@ -763,22 +772,27 @@ export class Placement {
             const hasText = numHorizontalGlyphVertices > 0 || numVerticalGlyphVertices > 0;
             const hasIcon = symbolInstance.numIconVertices > 0;
 
+            const placedOrientation = this.placedOrientations[symbolInstance.crossTileID];
+            const verticalHidden = (placedOrientation === WritingMode.horizontal || placedOrientation === WritingMode.horizontalOnly) ? 1 : 0;
+            const horizontalHidden = placedOrientation === WritingMode.vertical ? 1 : 0;
+
             if (hasText) {
                 const packedOpacity = packOpacity(opacityState.text);
                 // Vertical text fades in/out on collision the same way as corresponding
                 // horizontal text. Switch between vertical/horizontal should be instantaneous
-                const opacityEntryCount = (numHorizontalGlyphVertices + numVerticalGlyphVertices) / 4;
-                for (let i = 0; i < opacityEntryCount; i++) {
-                    bucket.text.opacityVertexArray.emplaceBack(packedOpacity);
+                const horizontalOpacity = horizontalHidden ? PACKED_HIDDEN_OPACITY : packedOpacity;
+                for (let i = 0; i < numHorizontalGlyphVertices / 4; i++) {
+                    bucket.text.opacityVertexArray.emplaceBack(horizontalOpacity);
+                }
+                const verticalOpacity = verticalHidden ? PACKED_HIDDEN_OPACITY : packedOpacity;
+                for (let i = 0; i < numVerticalGlyphVertices / 4; i++) {
+                    bucket.text.opacityVertexArray.emplaceBack(verticalOpacity);
```

Comment:
these are duplicated, could you make it into a function?

Reply:
yep, done in 0bacb7340909f8d674037d875e90e52ba1ea83fa!